File: Pod/TestClasses/OCSlimFitnesseTestReportReader.m
#import "OCSlimFitnesseTestReportReader.h"

@implementation OCSlimFitnesseTestReportFileReader

- (NSData *) read {
    
    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
    
    NSString *path = [bundle pathForResource:@"Fitnesse-Test-Report" ofType:@"xml"];
    
    NSAssert(path != nil, @"Fitnesse-Test-Report.xml file is missing. Check the setup of your Test target... (provide further details to make this a more useful message)");
    
    return [NSData dataWithContentsOfFile:path];
    
}

@end


/// OCSlimFitnesseTestReportCenter ///

@implementation OCSlimFitnesseTestReportCenter

static id<OCSlimFitnesseTestReportReader> _defaultReader;

+ (id<OCSlimFitnesseTestReportReader>)defaultReader {
    
    return _defaultReader;
    
}

+ (void)initialize {
    
    _defaultReader = [[OCSlimFitnesseTestReportFileReader alloc] init];
    
}

+ (void)setDefaultReader:(id<OCSlimFitnesseTestReportReader>)reader {
    
    _defaultReader = reader;
    
}


@end

File: Example/OCSlimProject_FitnesseTests/OCSlimFitnesseTestReportReader.swift
import Foundation

@objc class OCSlimFitnesseTestReportCenter: NSObject {
    
    static var defaultReader: OCSlimFitnesseTestReportReader  = OCSlimFitnesseTestReportFileReaderTests()
    
}

@objc protocol OCSlimFitnesseTestReportReader {
    
    func read() -> NSData
    
}

class OCSlimFitnesseTestReportFileReaderTests: NSObject, OCSlimFitnesseTestReportReader {
    
    func read() -> NSData {
        
        let bundle = NSBundle(forClass: OCSlimFitnesseTestReportFileReaderTests.self)
        
        let path = bundle.pathForResource("Fitnesse-Test-Report", ofType: "xml")!
        
        return NSData(contentsOfFile: path)!
        
    }
    
}

class OCSlimFitnesseTestReportReaderStub: NSObject, OCSlimFitnesseTestReportReader {
    
    private let data: NSData
    
    init(data: NSData) {
        self.data = data
        super.init()
    }
    
    func read() -> NSData {
        
        return data
        
    }
    
}